src/controllers/controller.bar.js

```diff
@@ -644,7 +646,9 @@ export default class BarController extends DatasetController {
         ? computeFlexCategoryTraits(index, ruler, options, stackCount * axisCount)
         : computeFitCategoryTraits(index, ruler, options, stackCount * axisCount);
 
-      const axisNumber = this._getAxis().indexOf(valueOrDefault(this.getDataset().xAxisID, this.getFirstScaleIdForXAxis()));
+      const axisNumber = this._getAxis().indexOf(valueOrDefault(
+        this.chart.options.indexAxis === 'x' ? this.getDataset().xAxisID : this.getDataset().yAxisID, this.getFirstScaleIdForIndexAxis()
```

Comment:
[nit] for readability I might pull the axis ID out into a separate variable

Reply:
done ! thanks for the suggestion